docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,292 @@
+
+## `@_distributedActorIndependent`
+
+## `@_disfavoredOverload`
+
+Marks a declaration that the type checker should try to avoid using. When the
+expression type checker is considering overloads, it will prefer a solution
+with fewer `@_disfavoredOverload` declarations over one with more of them.
+
+Historically, it was designed to help with `ExpressibleByXYZLiteral` types.
+The type checker strongly prefers to give literals their default type
+(e.g. `Int` for `ExpressibleByIntegerLiteral`,
+`String` for `ExpressibleByStringLiteral`, etc.).
+So if you want your API to prefer some other type but accept the default one
+too, you may need to mark the declaration taking the default literal type with
+this attribute:
+
+    extension LocalizedStringKey: ExpressibleByStringLiteral { ... }
+
+    extension Text {
+        // We want `Text("foo")` to use this initializer:
+        init(_ key: LocalizedStringKey) { ... }
+
+        // But without @_disfavoredOverload, it would use this one instead,
+        // because that lets it give the literal its default type:
+        @_disfavoredOverload init<S: StringProtocol>(_ str: S) { ... }
+    }
+
+`@_disfavoredOverload` is a stop-gap design; eventually the type-checker
+behavior should be fixed and the attribute should be removed/made into a no-op.
+
+It's main use case is to work around clearly undesirable overload resolution,
+not to arbitrarily manipulate overload priority.
+
+## `@_dynamicReplacement(for:)`
+
+Marks a function as the dynamic replacement for another `dynamic` function.
+This is similar to method swizzling in other languages such as Objective-C,
+except that the replacement happens at program start (or loading a shared
+library), instead of at an arbitrary point in time.
+
+For more details, see [Dynamic method replacement](https://forums.swift.org/t/dynamic-method-replacement/16619).
+
+## `@_effects`
+
+## `@_exported`
+
+Use to mark an imported module to re-export all its declarations.
+This attribute is most commonly used by overlays.
+
+```
+// module M
+func f() {}
+
+// module N
+@_exported import M
+
+// module P
+import N
+let () = N.f() // OK
+```
+
+## `@_fixed_layout`
+
+## `@_hasInitialValue`
+
+## `@_hasMissingDesignatedInitializers`
+
+## `@_hasStorage`
+
+## `@_implementationOnly`
+
+Used to mark an imported module as an implementation detail.
+This prevents types from that module being exposed in API
+(types of public functions, constraints in public extension etc.)
+and ABI (usage in `@inlinable` code).
+
+## `@_implements`
+
+## `@_implicitSelfCapture`
+
+Allows access to `self` inside a closure without explicitly capturing it,
+even when `Self` is a reference type.
+
+```
+class C {
+  func f() {}
+  func g(_: @escaping () -> ()) {
+    g({ f() }) // error: call to method 'f' in closure requires explicit use of 'self'
+  }
+  func h(@_implicitSelfCapture _: @escaping () -> ()) {
+    h({ f() }) // ok
+  }
+}
```

Comment:
```suggestion
```swift
class C {
  func f() {}
  func g(_: @escaping () -> Void) {
    g({ f() }) // error: call to method 'f' in closure requires explicit use of 'self'
  }
  func h(@_implicitSelfCapture _: @escaping () -> Void) {
    h({ f() }) // ok
  }
}
```